tlog-witness.md

```diff
@@ -0,0 +1,226 @@
+# The Witness Protocol
```

Comment:
+1, or `tlog-sync-witness` if fewer characters is the goal.

Reply:
(Should also add: question is a bit off topic -- just curious where you're at currently. Might be easier and more appropriate to sync outside of the PR!)

Reply:
As Al said, we already have a very similar API to this inside the AW: https://github.com/transparency-dev/witness/blob/main/api/http.go. This is not exposed by design, and is instead wrapped in AW to make it an asynchronous client-mode witness, instead of the synchronous server-mode witness that this spec describes.

We don't foresee a future where the AW will be a directly addressable server (though bastion is an option down the line), and thus _this spec_ won't describe the AW as anyone interacts with it. Hence our preference that this clearly says `sync` in the name. It would be an own-goal for transparency if this spec and the AW witness network comes out, and the AW doesn't implement the `tlog-witness` spec, and there is no network of witnesses implementing said spec.

Reply:
Sync vs async is a broader conversation that I think we should have in person (one could even say, synchronously :P). As you know, I am very skeptical that async witnesses can serve a diverse tlog ecosystem. (As a recent example, it would be much more complicated for APT repository signing to have to pause, expose tiles on HTTP somehow, poll, and then resume signing the repository. Compare with https://github.com/FiloSottile/litetlog/tree/main/cmd/apt-transport-tlog which wouldn't have to change basically to integrate synchronous witnesses.)

For the name, I think it only makes sense to have "sync" in it if async witnesses need a spec. My understanding is that AW would reach out to logs (for which we need a tiles spec) and then publish checkpoints (for which we already have checkpoint and cosignature specs). I guess we could document the endpoint at which the checkpoint is served, but that feels light for a spec, and more appropriate for an API documentation.